whylog/config/parser_name_manager.py

```diff
@@ -0,0 +1,44 @@
+import re
+
+
+class ParserNameManager(object):
+    def __init__(self, parsers):
+        self._parsers_name = set()
+        for parser_name in parsers:
+            self._parsers_name.add(parser_name)
+
+    def add_new_parser_name(self, parser_name):
+        self._parsers_name.add(parser_name)
+
+    def is_free_parser_name(self, parser_name, black_list):
+        return (parser_name not in self._parsers_name) and (parser_name not in black_list)
+
+    def _try_get_simple_name_else_append_number(self, word, black_list):
+        if self.is_free_parser_name(word, black_list):
+            return word
+        return self._find_free_by_number_appending(word, black_list)
+
+    def propose_parser_name(self, line, regex_str, black_list):
+        building_words = ParserNameManager._get_building_words(line, regex_str)
+        if not building_words:
+            return self._find_free_by_number_appending('parser_name', black_list)
+        if len(building_words) == 1:
+            self._try_get_simple_name_else_append_number(building_words[0], black_list)
+        proposed_name = building_words[0] + '_' + building_words[1]
```

Comment:
Maybe move this string joining into method `_try_get_simple_name_else_append_number`.
I think it is it's responsibility and code of propose_parser_name will be cleaner.
Consider function that takes: words_count, words_list and blacklist instead of word and black_list.
If it will be parametrized by words_count, we could then easily modify code when Client would prefer 3-words names rather than 2-words names.